### PR TITLE
Display already uploaded files in work form

### LINKED ISSUE #440: Improve positioning of "Choose files" widget in work form
The "Choose files" widget should render above uploaded files. I was able to reproduce this by coming back to the edit view after depositing a work with some files.

Screenshot from 2020-11-12 14-08-18

This is how it ought to look:

Screenshot from 2020-11-12 14-11-55

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made?

Fixes #440, #392 

See commit messages.

## How was this change tested?

CI and in browser locally.

### BEFORE

Screenshot from 2020-11-16 14-40-21

### AFTER
Screenshot from 2020-11-16 14-40-43

## Which documentation and/or configurations were updated?

None

### CODE REVIEW

File: spec/components/works/add_files_component_spec.rb [hunk 1]
```diff
@@ -3,12 +3,35 @@
...
+# rubocop:disable RSpec/AnyInstance
 RSpec.describe Works::AddFilesComponent do
   let(:form) { ActionView::Helpers::FormBuilder.new(nil, nil, controller.view_context, {}) }
   let(:rendered) { render_inline(described_class.new(form: form)) }
 
-  it 'renders the component' do
-    expect(rendered.to_html).to include('Add your files')
-    expect(rendered.css('button.dz-clickable').to_html).to include('Choose files')
+  before do
+    allow_any_instance_of(ActionView::Helpers::FormBuilder).to receive(:object).and_return(object)
```
Comment (jcoyne) [inline spec/components/works/add_files_component_spec.rb:13]: why don't you pass the object on line 9 rather than stubbing?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/components/works/add_files_component_spec.rb (hunk 1 above)
Comment (mjgiarlo) [reply spec/components/works/add_files_component_spec.rb:13]: I tried that and it didn't work. NoMethodError: Cannot call `object` on `nil`.

Comment (mjgiarlo) [reply spec/components/works/add_files_component_spec.rb:13]: The formbuilder instance that is created on line 9 is different than the one instantiated by the view component, apparently. They have different object IDs, which explains the NoMethodError. Hence the gross any_instance jazz.

@jcoyne :point_up:

File: spec/components/works/add_files_component_spec.rb [hunk 2]
```diff
@@ -3,12 +3,35 @@
...
+
+    it 'renders the component' do
+      expect(rendered.to_html).to include('Add your files')
+      expect(rendered.css('button.dz-clickable').to_html).to include('Choose files')
+    end
+  end
+
+  context 'with a persisted file component' do
+    let(:model) { create(:attached_file, :with_file) }
+    let(:object) { double('anonymous form instance', persisted?: true, model: model) }
```
Comment (jcoyne) [inline spec/components/works/add_files_component_spec.rb:27]:
```suggestion
  let(:form) { ActionView::Helpers::FormBuilder.new(nil, work_form, controller.view_context, {}) }
  let(:rendered) { render_inline(described_class.new(form: form)) }
  let(:work_form) { WorkForm.new(object) }

  context 'with an unpersisted file component' do
    let(:object) { build(:work) }

    it 'renders the component' do
      expect(rendered.to_html).to include('Add your files')
      expect(rendered.css('button.dz-clickable').to_html).to include('Choose files')
    end
  end

  context 'with a persisted file component' do
    let(:model) { create(:attached_file, :with_file) }
    let(:object) { build(:work, attached_files: [model]) }
```

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/components/works/add_files_component_spec.rb (hunk 2 above)
Comment (jcoyne) [reply spec/components/works/add_files_component_spec.rb:27]: How about this?

Comment (mjgiarlo) [reply spec/components/works/add_files_component_spec.rb:27]: Looks good. Looks like the trick is injecting a form as the "object" of the form builder? I'll patch it in. Thanks!

Comment (jcoyne) [reply spec/components/works/add_files_component_spec.rb:27]: yes.

Comment (mjgiarlo) [reply spec/components/works/add_files_component_spec.rb]: @jcoyne pushed. have another look?

**Review by jcoyne — APPROVED** [review]

**Pull request merged**